### PR TITLE
Modernize Python in gdal_tutorial

### PR DESCRIPTION
**Pull request by kannes**
- Modern importing from osgeo
- Safe key lookup in dict the simple .get() way
- format -> fileformat because format() is a built-in function
- print function instead of statement
- .format() method on strings instead of concatenation or %
- Some unpythonic things fixed, bit of PEP8ing like removal of many spaces
- Named keyword parameters on the more complex functions where available

### CODE REVIEW

File: gdal/doc/gdal_tutorial.dox [hunk 1]
```diff
@@ -155,16 +154,19 @@ In C:
...
+    print("Projection is {}".format(dataset.GetProjection()))
 
     geotransform = dataset.GetGeoTransform()
-    if not geotransform is None:
-	print 'Origin = (',geotransform[0], ',',geotransform[3],')'
-	print 'Pixel Size = (',geotransform[1], ',',geotransform[5],')'
+    if geotransform:
+        print("Origin = ({}, {})".format(geotransform[0], geotransform[3]))
+        print("Pixel Size = ({}, {})".format(geotransform[1], geotransform[5]))
 \endcode
```
Comment (kannes) [inline gdal/doc/gdal_tutorial.dox]: Ideally I wanted to use some descriptive names here, e.g. `print("Pixel Size = ({xsize}, {ysize})".format(xsize=geotransform[1], ysize=geotransform[5]))` but I was not sure what the elements of geotransform all really are so skipped that thought.

File: gdal/doc/gdal_tutorial.dox [hunk 2]
```diff
@@ -417,15 +420,15 @@ In C:
...
+    driver = gdal.GetDriverByName(fileformat)
     metadata = driver.GetMetadata()
-    if metadata.has_key(gdal.DCAP_CREATE) \
-       and metadata[gdal.DCAP_CREATE] == 'YES':
-        print 'Driver %s supports Create() method.' % format
-    if metadata.has_key(gdal.DCAP_CREATECOPY) \
-       and metadata[gdal.DCAP_CREATECOPY] == 'YES':
-        print 'Driver %s supports CreateCopy() method.' % format
+
+    if metadata.get(gdal.DCAP_CREATE) == "YES":
```
Comment (kannes) [inline gdal/doc/gdal_tutorial.dox]: Using dict.get(key) returns either the value or None (~=False), this makes the condition much simpler.

**Comment by avalentino** [comment]
+1

Probably I would also add

```
from __future__ import print_function
```

at the top of the script to have 100% compatibility with Python 2

**Comment by kannes** [comment]
It's not a single script but snippets are more or less independent. Not beautiful but I didn't design it that way ;)

File: gdal/doc/gdal_tutorial.dox [hunk 3]
```diff
@@ -57,11 +57,10 @@ int main()
 
 In Python:
 \code
-    import gdal
-    from gdalconst import *
+    from osgeo import gdal, gdalconst
 
-    dataset = gdal.Open( filename, GA_ReadOnly )
-    if dataset is None:
+    dataset = gdal.Open(filename, gdalconst.GA_ReadOnly)
```
Comment (rouault) [inline gdal/doc/gdal_tutorial.dox]: personnaly, I never use gdalconst but just gdal. So gdal.GA_ReadOnly. That's probably easier for users to use only the gdal module

Comment (kannes) [inline gdal/doc/gdal_tutorial.dox]: I blindly took stuff from the original where gdalconst was imported and I assumed it was just for this constant. Your way sounds much more reasonable. Should I edit it or how does that work? :)

Comment (rouault) [inline gdal/doc/gdal_tutorial.dox]: Yes, please add a new commit into this branch. I'll get the global diff before applying to SVN

Comment (kannes) [inline gdal/doc/gdal_tutorial.dox]: Done!

**Comment by rouault** [comment]
Committed into r39831. Thanks

**Pull request closed without merging**